### PR TITLE
Minor changes to the environment file and cleanup

### PR DESCRIPTION
**Pull request by epassaro**
## Description
- Remove `codecov` dependency from `tardis_env.yml` since is not required to make Codecov work.
- Remove `pytest-cov` line from test pipeline since it's already installed by the environment file.
- Remove old `readthedocs.yml` files.

## Motivation and Context

Tidying up.

## How Has This Been Tested?

Test and codecov report work.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have assigned/requested two reviewers for this pull request.

### CODE REVIEW

File: azure-pipelines/simple_test_framework.yml [hunk 1]
```diff
@@ -55,7 +53,6 @@ jobs:
 
       - bash: |
           source activate tardis
-          conda install -y pytest-cov
```
Comment (epassaro) [inline azure-pipelines/simple_test_framework.yml:58]: `pytest-cov` is already installed in the environment.

File: tardis_env3.yml [hunk 2]
```diff
@@ -57,16 +55,14 @@ dependencies:
   - git-lfs
   - pytest=5
   - pytest-html
-  - requests
+  - pytest-cov
   - coverage
+  - requests
   - docopt
-  - pytest-cov
-  - codecov
```
Comment (epassaro) [inline tardis_env3.yml:64]: `codecov` package is not necessary to make Codecov work.

File: tardis_env3.yml [hunk 3]
```diff
@@ -10,11 +10,9 @@ dependencies:
   - scipy=1.1
   - pandas=0.24
   - astropy=3
-
   - numba=0.43
   - numexpr
   - Cython=0.29
```
Comment (jaladh-singhal) [inline tardis_env3.yml:15]: I'm not sure about this - if these packages are related somehow (I guess that's why there are newlines), we can add a comment at top to mention the purpose?

Comment (epassaro) [reply tardis_env3.yml:15]: I think that was the VSCode YAML extension/autoformatter.

**Pull request merged**